sickbeard/webserve.py

```diff
@@ -1417,13 +1418,12 @@ def pickManualSnatch(self, show=None, season=None, episode=None, provider=None,
 
         sickbeard.searchQueueScheduler.action.add_item(ep_queue_item)
 
-        if not ep_queue_item.started and ep_queue_item.success is None:
-            return json.dumps(
-                {'result': 'success'})  # I Actually want to call it queued, because the search hasnt been started yet!
-        if ep_queue_item.started and ep_queue_item.success is None:
-            return json.dumps({'result': 'success'})
-        else:
-            return json.dumps({'result': 'failure'})
+        while ep_queue_item.success is not False:
```

Comment:
The queue_item will return None until something goes wrong (False) or the snatch succeeds (True). Maybe there are exceptions, but haven't been able to reproduce those.